src/vcpkg/base/downloads.cpp

```diff
@@ -970,25 +984,29 @@ namespace vcpkg
         {
             if (urls.size() != 0)
             {
+                msg::println(LocalizedString::from_raw(fmt::format("Downloading {}", urls[0])));
+                
                 auto maybe_url = try_download_file(
                     fs, urls, headers, download_path, sha512, m_config.m_secrets, errors, progress_sink);
                 if (auto url = maybe_url.get())
                 {
                     if (auto hash = sha512.get())
                     {
-                        auto maybe_push = put_file_to_mirror(fs, download_path, *hash);
-                        if (!maybe_push)
+                        if(m_config.m_write_url_template.has_value())
                         {
-                            msg::println_warning(msgFailedToStoreBackToMirror);
-                            msg::println(maybe_push.error());
-                        }
+                            put_file_to_mirror(fs, download_path, *hash);
+                            msg::println(msgAssetCacheSuccesfullyStored, msg::path = download_path.filename());
+                        
+                        }else{
+                            msg::println_warning(msgFailedToStoreBackToMirror, msg::path = download_path.filename());
```

Comment:
This message should still depend on the result of `put_file_to_mirror()`, shouldn't it?

Reply:
Yes